core/controllers/access_validators_test.py

```diff
@@ -416,3 +418,72 @@ def test_validation_returns_false_if_given_user_is_non_existent(
             ), expected_status_int=404
         )
         self.logout()
+
+
+class TopicEditorPageAccessValidationHandlerTests(test_utils.EmailTestBase):
+    """Checks the access to the topic editor page and its rendering."""
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.signup(self.CURRICULUM_ADMIN_EMAIL, self.CURRICULUM_ADMIN_USERNAME)
+        self.add_user_role(
+            self.CURRICULUM_ADMIN_USERNAME, feconf.ROLE_ID_CURRICULUM_ADMIN)
+
+        self.admin_id = self.get_user_id_from_email(self.CURRICULUM_ADMIN_EMAIL)
+
+        self.skill_id = skill_services.get_new_skill_id()
+        self.save_new_skill(
+            self.skill_id, self.admin_id, description='Skill Description')
+        self.skill_id_2 = skill_services.get_new_skill_id()
+        self.save_new_skill(
+            self.skill_id_2, self.admin_id, description='Skill Description 2')
+        self.topic_id = topic_fetchers.get_new_topic_id()
+        self.save_new_topic(
+            self.topic_id, self.admin_id, name='Name',
+            abbreviated_name='topic-one', url_fragment='topic-one',
+            description='Description', canonical_story_ids=[],
+            additional_story_ids=[],
+            uncategorized_skill_ids=[self.skill_id, self.skill_id_2],
+            subtopics=[], next_subtopic_id=1)
+
+    def test_access_topic_editor_page_without_logging_in(self) -> None:
+        self.get_json(
+            '%s/can_access_topic_editor/%s' % (
+            ACCESS_VALIDATION_HANDLER_PREFIX,
+            self.topic_id), expected_status_int=401)
+
+    def test_access_topic_editor_page_with_guest_user(self) -> None:
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+        self.login(self.VIEWER_EMAIL)
+        self.get_json(
+            '%s/can_access_topic_editor/%s' % (
+            ACCESS_VALIDATION_HANDLER_PREFIX,
+            self.topic_id), expected_status_int=401)
+        self.logout()
+
+        self.login(self.NEW_USER_EMAIL)
+        self.get_json(
+            '%s/can_access_topic_editor/%s' % (
+                ACCESS_VALIDATION_HANDLER_PREFIX, self.topic_id),
+            expected_status_int=401)
+        self.logout()
+
+    def test_access_topic_editor_page_with_curriculum_admin(
+        self
+    ) -> None:
+        self.login(self.CURRICULUM_ADMIN_EMAIL)
+        self.get_html_response(
+            '%s/can_access_topic_editor/%s' % (
+                ACCESS_VALIDATION_HANDLER_PREFIX,
+                self.topic_id), expected_status_int=200)
+        self.logout()
+
+    def test_cannot_access_topic_editor_page_with_invalid_topic_id(
```

Comment:
@DubeySandeep 
here is the test for invalid_id

Reply:
Hi @DubeySandeep 

```
def test_cannot_access_topic_editor_page_with_invalid_topic_id(
        self
    ) -> None:
        self.login(self.NEW_USER_EMAIL)
        self.get_json(
            '%s/can_access_topic_editor/%s' % (
                ACCESS_VALIDATION_HANDLER_PREFIX,
                topic_fetchers.get_new_topic_id()), expected_status_int=401)
        self.logout()
```

changed `invalid_id` to `topic_fetchers.get_new_topic_id()` for valid id 
still coverage test is failing

![Screenshot 2024-02-19 at 2 26 10 PM](https://github.com/oppia/oppia/assets/108923011/b2c05575-a6e4-43ff-a0f3-069387256ad2)

Reply:
This is quite obviously not hitting the line in question. It's raising a 401 and not a 404 (which would correspond to PageNotFoundException). You'll need to do some analysis -- e.g. perhaps this case is already caught by the ACL decorator?

Reply:
Hmmm, right. thanks
updated the tests